types/intset.go

```diff
@@ -0,0 +1,60 @@
+package types
+
+import (
+	"github.com/golang-collections/go-datastructures/bitarray"
+)
+
+// IntSet is a space-efficient set of uint64
+type IntSet struct {
+	ba bitarray.BitArray
+}
+
+// NewIntSet returns an new IntSet, optionally initialized with integers
+func NewIntSet(ints ...uint64) IntSet {
+	out := IntSet{ba: bitarray.NewSparseBitArray()}
+	for _, i := range ints {
+		out.Add(i)
+	}
+	return out
+}
+
+// Add adds an integer to this IntSet
+func (is IntSet) Add(i uint64) {
+	// Ignoring errors as we are using SparseBitArrays, which never return errors
+	_ = is.ba.SetBit(i)
+}
+
+// Contains returns whether an integer exists in this IntSet
+func (is IntSet) Contains(i uint64) bool {
```

Comment:
`Contains` should operate on another IntSet (this wasn't clear from the design doc).  If you are interested in the motivation check out `submitPoSt` and the `ValidateFaultSet` specs here: https://github.com/filecoin-project/specs/blob/master/actors.md#submitpost